File: Brasserie_mav/src/main/java/model/Ingredient.java
package model;

import java.util.List;

import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import javax.persistence.ManyToMany;
import javax.persistence.Version;

@Entity
public class Ingredient {
	
	@Id
	@GeneratedValue(strategy = GenerationType.IDENTITY)
	private Integer id_ingredient;
	private String nom;
	private int stock = 0;
	
	@ManyToMany(mappedBy= "recette")
	private List<Biere> bieres;
	
	@Version
	protected int version;
	
	// contructors
	public Ingredient() {}
	
	public Ingredient(String nom, int stock) {
		this.nom = nom;
		this.stock = stock;
	}
	
	public Ingredient(String nom) {
		this.nom = nom;
	}
	
	public Ingredient(String nom, int stock, List<Biere> bieres)
	{
		this.nom = nom;
		this.stock = stock;
		this.bieres = bieres;
	}
	

	public Integer getId_ingredient() {
		return id_ingredient;
	}

	public void setId_ingredient(Integer id_ingredient) {
		this.id_ingredient = id_ingredient;
	}

	public String getNom() {
		return nom;
	}

	public void setNom(String nom) {
		this.nom = nom;
	}

	public int getStock() {
		return stock;
	}

	public void setStock(int stock) {
		this.stock = stock;
	}

	public List<Biere> getBieres() {
		return bieres;
	}

	public void setBiere(List<Biere> bieres) {
		this.bieres = bieres;
	}

	
	
	// methods
	
}

File: Angular/Brasserie/src/app/model/ingredient.ts
import { Biere } from './biere';
export class Ingredient {
  public constructor(
    private _id_ingredient?: number | undefined,
    private _nom?: string | undefined,
    private _stock: number = 0,
    private _bieres?: Biere[] | undefined
  ) {}

  public get id_ingredient(): number | undefined {
    return this._id_ingredient;
  }

  public set id_ingredient(value: number | undefined) {
    this._id_ingredient = value;
  }

  public get nom(): string | undefined {
    return this._nom;
  }

  public set nom(value: string | undefined) {
    this._nom = value;
  }



  public get stock(): number  {
    return this._stock;
  }

  public set stock(value: number) {
    this._stock = value;
  }


public get bieres(): Biere []| undefined {
    return this._bieres;
  }

  public set bieres(value : Biere []| undefined) {
    this._bieres = value;
  }














}
